lib/core/stores/remote.ts

```diff
@@ -253,40 +309,177 @@ export class RemotePromiseStore implements IPromiseStore {
+          id,
+          description,
+          cron,
+          tags,
+          promiseId,
+          promiseTimeout,
+          promiseHeaders,
+          promiseData: promiseData ? encode(promiseData, this.encoder) : undefined,
+          promiseTags,
+        }),
+      },
+      3,
+      this.logger,
+    );
+
+    return schedule;
+  }
+
+  async get(id: string): Promise<Schedule> {
+    const schedule = call(
+      `${this.url}/schedules/${id}`,
+      isSchedule,
+      {
         headers: {
           Accept: "application/json",
           "Content-Type": "application/json",
         },
         method: "GET",
-      });
+      },
+      3,
+      this.logger,
+    );
 
-      cursor = res.cursor;
-      yield res.promises.map((p) => this.decode(p));
-    }
+    return schedule;
   }
 
-  private encode(value: string): string {
-    try {
-      return this.encoder.encode(value);
-    } catch (e: unknown) {
-      throw new ResonateError(ErrorCodes.ENCODER, "Encode error", e);
-    }
+  async delete(id: string): Promise<boolean> {
+    await call(
+      `${this.url}/schedules/${id}`,
+      this.isDeletedResponse,
+      {
+        method: "DELETE",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+      },
+      3,
+      this.logger,
+    );
+    return true;
   }
 
-  private decode<P extends DurablePromise>(promise: P): P {
-    try {
-      if (promise.param?.data) {
-        promise.param.data = this.encoder.decode(promise.param.data);
+  async search(
+    id: string,
+    tags?: Record<string, string>,
+    limit?: number,
+    cursor?: string | undefined,
+  ): Promise<{ cursor: string; schedules: Schedule[] }> {
+    const params = new URLSearchParams({ id });
+
+    if (tags) {
+      for (const [k, v] of Object.entries(tags)) {
+        params.append(`tags[${k}]`, v);
       }
+    }
 
-      if (promise.value?.data) {
-        promise.value.data = this.encoder.decode(promise.value.data);
-      }
+    if (limit !== undefined) {
+      params.append("limit", limit.toString());
+    }
 
-      return promise;
-    } catch (e: unknown) {
-      throw new ResonateError(ErrorCodes.ENCODER, "Decode error", e);
+    if (cursor !== undefined) {
+      params.append("cursor", cursor);
     }
+
+    const url = new URL(`${this.url}/schedules`);
+    url.search = params.toString();
+
+    const result = await call(
+      url.toString(),
+      isSearchSchedulesResult,
+      {
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+        method: "GET",
+      },
+      3,
+      this.logger,
+    );
+
+    return {
+      cursor: result.cursor,
+      schedules: result.schedules.map((schedule: Schedule) => schedule),
+    };
+  }
+
+  // any response from delete is true
+  private isDeletedResponse(_: any): _ is any {
+    return true;
+  }
+}
+
+export class RemoteStore implements IStore {
```

Comment:
Let's take a similar approach here and have `RemoteStore` accept a IPromiseStore and IScheduleStore as constructor parameters.